src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp

```diff
@@ -115,6 +114,72 @@ class OpenGLDrawingContext final : public IDrawingContext
     void HandleTransparency();
 
     void SetDPI(rct_drawpixelinfo* dpi);
+    rct_drawpixelinfo* GetDPI() const
+    {
+        return _dpi;
+    }
+};
+
+class OpenGLRainDrawer final : public IRainDrawer
+{
+    OpenGLDrawingContext* _drawingContext;
+
+public:
+    explicit OpenGLRainDrawer(OpenGLDrawingContext* drawingContext)
+        : _drawingContext(drawingContext)
+    {
+    }
+
+    virtual void Draw(int32_t x, int32_t y, int32_t width, int32_t height, int32_t xStart, int32_t yStart)
+    {
+        // clang-format off
+        static constexpr const uint8_t RainPattern[] =
+        {
+            32, 32, 0, 12, 0, 14, 0, 16, 255, 0, 255, 0, 255, 0, 255, 0, 255,
+            0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0,
+            255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255,
+            0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 255, 0, 0, 0
+        };
+        // clang-format on
```

Comment:
You should share this data between the two rain drawers.

Reply:
I wasn't sure where to put it exactly as the rain interface is supposed to be abstract, I could introduce a header file like RainPatterns.h if thats alrite

Reply:
Rain.h and Rain.cpp is where I would put it.

Reply:
Good point, its oddly abstract in too many ways.